### PR TITLE
CMake Refactoring

### PR DESCRIPTION
**Pull request by mgaudet**
Some small changes to the CMakeLists that enable future changes.

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@rwy0717 , @dnakamura I'd appreciate a review from you as well on this PR.

**Comment by dnakamura** [comment]
lgtm

File: cmake/versions.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
+###############################################################################
+#
+# (c) Copyright IBM Corp. 2017
```
Comment (0xdaryl) [inline cmake/versions.cmake]: 2017, 2017

File: cmake/vpath.cmake [hunk 2]
```diff
@@ -0,0 +1,50 @@
+###############################################################################
+#
+# (c) Copyright IBM Corp. 2017
```
Comment (0xdaryl) [inline cmake/vpath.cmake]: 2017, 2017

File: cmake/versions.cmake [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+#TODO figure out method of sharing version info with makefile build
+set(OMR_VERSION_MAJOR 0)
+set(OMR_VERSION_MINOR 0)
+set(OMR_VERSION_PATCH 1)
+set(OMR_VERSION ${OMR_VERSION_MAJOR}.${OMR_VERSION_MINOR}.${OMR_VERSION_PATCH})
+
+
+set(OMR_VERSION_STRING "<Unknown>" CACHE STRING "") # TODO: Set the version
+set(OMR_JIT_VERSION_STRING "<Unknown>" CACHE STRING "") # TODO: Set the version
```
Comment (0xdaryl) [inline cmake/versions.cmake]: I'm not a big fan of TODOs in general.  In this context they're fairly obvious TODOs aren't they?

Comment (mgaudet) [inline cmake/versions.cmake]: So, they aren't new TODOs, they were pulled in from the previous prototype CMake work. I'll remove them tho.

**Pull request merged**